erdos/src/dataflow/operators/map.rs

```diff
@@ -10,77 +10,110 @@ use crate::dataflow::{
 /// function.
 ///
 /// # Example
-/// The below example shows how to use a MapOperator to double an incoming stream of usize
+/// The below example shows how to use a [`FlatMapOperator`] to double an incoming stream of usize
 /// messages, and return them.
 ///
 /// ```
-/// # use erdos::dataflow::{stream::IngestStream, operator::{OperatorConfig}, operators::{MapOperator}};
+/// # use erdos::dataflow::{stream::IngestStream, operator::{OperatorConfig}, operators::{FlatMapOperator}};
 /// # let source_stream = IngestStream::new();
 /// let map_stream = erdos::connect_one_in_one_out(
-///     || -> MapOperator<usize, usize> { MapOperator::new(|a: &usize| -> usize { 2 * a }) },
+///     || -> FlatMapOperator<usize, _> {
+///         FlatMapOperator::new(|x: &usize| -> Vec<usize> { vec![2 * x] })
+///     },
 ///     || {},
-///     OperatorConfig::new().name("MapOperator"),
+///     OperatorConfig::new().name("FlatMapOperator"),
 ///     &source_stream,
 /// );
 /// ```
-pub struct MapOperator<D1, D2>
+pub struct FlatMapOperator<D, I>
 where
-    D1: Data + for<'a> Deserialize<'a>,
-    D2: Data + for<'a> Deserialize<'a>,
+    D: Data + for<'a> Deserialize<'a>,
+    I: IntoIterator,
+    I::Item: Data + for<'a> Deserialize<'a>,
 {
-    map_function: Arc<dyn Fn(&D1) -> D2 + Send + Sync>,
+    flat_map_fn: Arc<dyn Fn(&D) -> I + Send + Sync>,
 }
 
-impl<D1, D2> MapOperator<D1, D2>
+impl<D, I> FlatMapOperator<D, I>
 where
-    D1: Data + for<'a> Deserialize<'a>,
-    D2: Data + for<'a> Deserialize<'a>,
+    D: Data + for<'a> Deserialize<'a>,
+    I: IntoIterator,
+    I::Item: Data + for<'a> Deserialize<'a>,
 {
-    pub fn new<F>(map_function: F) -> Self
+    pub fn new<F>(flat_map_fn: F) -> Self
     where
-        F: 'static + Fn(&D1) -> D2 + Send + Sync,
+        F: 'static + Fn(&D) -> I + Send + Sync,
     {
         Self {
-            map_function: Arc::new(map_function),
+            flat_map_fn: Arc::new(flat_map_fn),
         }
     }
 }
 
-impl<D1, D2> OneInOneOut<(), D1, D2> for MapOperator<D1, D2>
+impl<D, I> OneInOneOut<(), D, I::Item> for FlatMapOperator<D, I>
 where
-    D1: Data + for<'a> Deserialize<'a>,
-    D2: Data + for<'a> Deserialize<'a>,
+    D: Data + for<'a> Deserialize<'a>,
+    I: IntoIterator,
+    I::Item: Data + for<'a> Deserialize<'a>,
 {
-    fn on_data(&mut self, ctx: &mut OneInOneOutContext<(), D2>, data: &D1) {
-        let timestamp = ctx.get_timestamp().clone();
-        ctx.get_write_stream()
-            .send(Message::new_message(timestamp, (self.map_function)(data)))
-            .unwrap();
-        tracing::debug!(
-            "{} @ {:?}: received {:?} and sent {:?}",
-            ctx.get_operator_config().get_name(),
-            ctx.get_timestamp(),
-            data,
-            (self.map_function)(data)
-        );
+    fn on_data(&mut self, ctx: &mut OneInOneOutContext<(), I::Item>, data: &D) {
+        tracing::info!("on data");
```

Comment:
Any particular reason this was kept separately and on `info!(...)`?

Reply:
Sorry, those are debug statements I forgot to remove.